rmgpy/rmg/main.py

```diff
@@ -1234,23 +1236,31 @@ def check_model(self):
         else:
             logging.info("No collision rate violators found.")
 
-    def make_seed_mech(self, first_time=False):
+    def initialize_seed_mech(self):
         """
-        causes RMG to make a seed mechanism out of the current chem_annotated.inp and species_dictionary.txt
-        this seed mechanism is outputted in a seed folder within the run directory and automatically
-        added to as the (or replaces the current) 'Seed' thermo and kinetics libraries in database
-        
-        if run with first_time=True it will change self.name to be unique within the thermo/kinetics libraries
-        by adding integers to the end of the name to prevent overwritting
+        Initialize the process of saving the seed mechanism by performing the following:
 
-        This also writes the filter tensors to the `filters` sub-folder for restarting an RMG job from a seed mechanism
+        1. Create the initial seed mechanism folder (the seed from a previous iterations will be deleted)
+        2. Save the restart-from-seed file (unless the current job is itself a restart job)
+        3. Ensure that we don't overwrite existing libraires in the database that have the same name as this job
         """
+        # Make the initial seed mechanism folder
+        seed_dir = os.path.join(self.output_directory, 'seed')
+        if os.path.exists(seed_dir):  # This is a seed from a previous RMG run. Delete it
+            shutil.rmtree(seed_dir)
+        os.mkdir(seed_dir)
 
-        logging.info('Making seed mechanism...')
+        # Generate a file for restarting from a seed mechanism if this is not a restart job
+        if not self.restart:
+            with open(os.path.join(self.output_directory, 'restart_from_seed.py'), 'w') as f:
+                f.write('restartFromSeed(path=\'seed\')\n\n')
+                with open(self.input_file, 'r') as input_file:
+                    f.write(''.join(input_file.readlines()))
```

Comment:
Could you do `f.write(input_file.read())`?

Reply:
Done, thanks for the catch!